deeplearning4j-core/src/main/java/org/deeplearning4j/util/StringCluster.java

```diff
@@ -87,9 +87,13 @@ public int compare(Map<String,Integer> o1, Map<String,Integer> o2) {
                 for (int i : o2.values()) {
                     total2 += i;
                 }
-                return total2 - total1;
-            } else {
-                return s2 - s1;
+                if (total2 < total1) return -1;
```

Comment:
I think overflow is a real potential concern here, enough that subtraction is probably a potential bug